azure-quantum/azure/quantum/job/base_job.py

```diff
@@ -0,0 +1,240 @@
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+##
+import abc
+import logging
+import uuid
+
+from urllib.parse import urlparse
+from typing import Any, Dict, TYPE_CHECKING
+from urllib.parse import urlparse
+from azure.storage.blob import BlobClient
+
+from azure.quantum.storage import create_container_using_client, get_container_uri, upload_blob, download_blob, ContainerClient
+from azure.quantum._client.models import JobDetails
+
+
+if TYPE_CHECKING:
+    from azure.quantum.workspace import Workspace
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT = 100
+
+
+class BaseJob(abc.ABC):
+    # Optionally override these to create a Provider-specific Job subclass
```

Comment:
What is this `abc.ABC`?

Reply:
It's an abstract base class, this means you can't instantiate an object from this class (you have to use a subclass). For more info see: https://docs.python.org/3/library/abc.html